### PR TITLE
Fixing the SPN broken demo and formating the worklet demo

### PR DESCRIPTION
**Pull request by TenghuiZhang**
Fixing #308

### CODE REVIEW

File: src/audio-worklet/migration/spn-recorder/app.js [hunk 1]
```diff
@@ -35,30 +42,30 @@ async function init() {
...
-      // 10 seconds seems reasonable for demo purposes.
-      context.sampleRate * 10,
-      context.sampleRate,
-  );
+  // Prepare max recording buffer for recording.
+  const recordingProperties = {
+    numberOfChannels: 2,
+    sampleRate: context.sampleRate,
+    maxFrameCount: context.sampleRate * 300
+  };
```
Comment (hoch) [inline src/audio-worklet/migration/spn-recorder/app.js:56]: This is a lot easier to read! 👍

File: src/audio-worklet/migration/spn-recorder/app.js [hunk 2]
```diff
@@ -70,17 +77,21 @@ async function init() {
...
+function setupScriptProcessor(recordingProperties, passSampleToVisualizers) {
   const processor = context.createScriptProcessor(BUFFER_SIZE);
-  const currentSamples = new Array(recordBuffer.numberOfChannels).fill([]);
+  const currentSamples =
+     new Array(recordingProperties.numberOfChannels).fill([]);
 
   // Main SPN callback. Handles recording data and tracking recording length.
   processor.onaudioprocess = function(event) {
+    // Display current recording length.
+    document.querySelector('#data-len').innerHTML =
```
Comment (hoch) [inline src/audio-worklet/migration/spn-recorder/app.js:93]: `innerHTML` to `textContent`

File: src/audio-worklet/migration/spn-recorder/app.js [hunk 3]
```diff
@@ -70,17 +77,21 @@ async function init() {
...
+    Math.round(recordingLength / recordingProperties.sampleRate * 100)/100;
```
Comment (hoch) [inline src/audio-worklet/migration/spn-recorder/app.js:94]: This line needs more indentation?

File: src/audio-worklet/migration/spn-recorder/app.js [hunk 4]
```diff
@@ -90,8 +101,9 @@ function setupScriptProcessor(recordBuffer, passSampleToVisualizers) {
 
       // While recording, feed data to recording buffer at the proper time.
       if (isRecording) {
-        recordBuffer
-            .copyToChannel(currentSamples[channel], channel, recordingLength);
+        let frameNumber = recordingLength / BUFFER_SIZE;
```
Comment (hoch) [inline src/audio-worklet/migration/spn-recorder/app.js:104]: With this operation, `frameNumber` can be a float. Is that okay?

File: src/audio-worklet/migration/spn-recorder/app.js [hunk 5]
```diff
@@ -111,37 +138,31 @@ function setupScriptProcessor(recordBuffer, passSampleToVisualizers) {
...
  */
-function setupRecording(recordBuffer) {
-  const recordButton = document.querySelector('#record');
-  const recordText = recordButton.querySelector('span');
-  const player = document.querySelector('#player');
-  const downloadButton = document.querySelector('#download');
-
-  async function prepareClip() {
+function setupRecording(recordingProperties) {
+  async function prepareClip(finalRecordBuffer) {
```
Comment (hoch) [inline src/audio-worklet/migration/spn-recorder/app.js:144]: Can we take this function out of `setupRecording` and add comments?

File: src/audio-worklet/migration/spn-recorder/app.js [hunk 6]
```diff
@@ -305,3 +326,26 @@ function setupRecordingGainVis() {
 
   return draw;
 }
+
+/**
+ * Create the recording buffer with right size
+ * @param {object} recordingProperties Properties of record buffer
+ * @return {AudioBuffer} Record buffer for current recording
```
Comment (hoch) [inline src/audio-worklet/migration/spn-recorder/app.js:333]: `@returns`

https://jsdoc.app/tags-returns.html

**Review by hoch — COMMENTED** [review]
Thanks for your patience! We're almost there.

File: src/audio-worklet/migration/spn-recorder/app.js (hunk 4 above)
Comment (TenghuiZhang) [reply src/audio-worklet/migration/spn-recorder/app.js:104]: JS don't have the different number type like int, float, or double. The only type is number. 

Based on my design, the recordingLength is increased by BUFFER_SIZE each time (line 115). Therefore, recordingLength/BUFFER_SIZE will always be an "INTEGER"

File: src/audio-worklet/migration/spn-recorder/app.js (hunk 5 above)
Comment (TenghuiZhang) [reply src/audio-worklet/migration/spn-recorder/app.js:144]: Yes, it is moved.

File: src/audio-worklet/migration/spn-recorder/app.js (hunk 4 above)
Comment (hoch) [reply src/audio-worklet/migration/spn-recorder/app.js:104]: > JS don't have the different number type like int, float, or double. The only type is number.

Yes. I am well aware of that. (and that can be a "bad part" for JS audio programming) The reason why I am asking this is that when this `frameNumber` becomes a floating point number it creates a confusion when we use this to pinpoint a sample value in the array. 

```js
a = [1, 2, 3]
a[0.75] // undefined
```

The resulting value of `undefined` will create `NaN` values in the audio stream for subsequent operation. (That's very bad.) So it is a general practice in audio programming to make it clear that the sample index number is a positive integer.

> Based on my design, the recordingLength is increased by BUFFER_SIZE each time (line 115). Therefore, recordingLength/BUFFER_SIZE will always be an "INTEGER"

You have two options:
1) Add a comment that explains why this operation is safe for sample indexing.
2) Use `Math.floor()` to make sure it falls on to an integer.

Does it make sense?

**Review by hoch — APPROVED** [review]
LGTM!

**Pull request merged**